lib/metricAggregators.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const util = require('./util');
+
+/**
+ * Returns a new function that applies the `aggregatorFn` to the values.
+ * @param {Function} aggregatorFn function to apply to values.
+ * @return {Function} aggregator function
+ */
+const AggregatorFactory = function(aggregatorFn) {
+	return metrics => {
+		if (metrics.length === 0) return;
+		const result = {
+			help: metrics[0].help,
+			name: metrics[0].name,
+			type: metrics[0].type,
+			values: []
+		};
+		// Gather metrics by labels.
+		const byLabels = new Map();
+		metrics.forEach(metric => {
+			metric.values.forEach(value => {
+				const key = util.hashObject(value.labels);
+				if (byLabels.has(key)) {
+					byLabels.get(key).push(value);
+				} else {
+					byLabels.set(key, [value]);
+				}
+			});
+		});
+		// Apply aggregator function to gathered metrics.
+		byLabels.forEach(values => {
+			if (values.length === 0) return;
+			const valObj = {
+				value: aggregatorFn(values),
+				labels: values[0].labels
+			};
+			if (values[0].metricName) {
+				valObj.metricName = values[0].metricName;
+			}
+			// TODO how do we aggregate timestamps? Average? Omit?
+			result.values.push(valObj);
+		});
+		return result;
+	};
+};
+
+/**
+ * Functions that can be used to aggregate metrics from multiple registries.
+ */
+exports.AGGREGATORS = {
```

Comment:
Im thinking these might belong in a separate file? Not sure about the uppercase function names either